File: apps/android/app/src/main/java/com/example/shop/feature/product/data/model/Product.kt
package com.example.shop.feature.product.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ProductListRequest(
    val page: Int = 1,
    val pageSize: Int = 20,
    val sort: String = "createdAt",
    val order: String = "desc",
    val filters: ProductFilters? = null,
)

@Serializable
data class ProductFilters(
    val status: String? = null,
    val categoryId: String? = null,
    val brand: String? = null,
)

@Serializable
data class ProductListItem(
    val id: String,
    val title: String,
    val slug: String,
    val brand: String? = null,
    val status: String,
    val minPrice: String? = null,
    val maxPrice: String? = null,
    val totalSales: Int,
    val avgRating: String,
    val reviewCount: Int,
    val primaryImage: String? = null,
    val createdAt: String,
)

File: apps/ios/Shop/Feature/Product/Data/Model/Product.swift
import Foundation
import ComposableArchitecture

struct Product: Codable, Equatable, Identifiable, Sendable {
    let id: String
    let title: String
    let slug: String
    let brand: String?
    let status: String
    let minPrice: String?
    let maxPrice: String?
    let totalSales: Int
    let avgRating: String
    let reviewCount: Int
    let primaryImage: String?
    let createdAt: String
}
